addons/l10n_be/models/res_partner.py

```diff
@@ -12,9 +12,12 @@ class ResPartner(models.Model):
     @api.depends('vat', 'country_id')
     def _compute_company_registry(self):
         # OVERRIDE
-        # If a belgian company has a VAT number then it's company registry is it's VAT Number (without country code).
+        # If a belgian company has a VAT number then its company registry is its VAT Number (without country code).
         super()._compute_company_registry()
         for partner in self.filtered(lambda p: p.country_id.code == 'BE' and p.vat):
             vat_country, vat_number = self._split_vat(partner.vat)
+            if vat_country.isnumeric():
+                vat_country = partner.country_code.lower()
```

Comment:
```suggestion
                vat_country = 'be' 
``` 
The country seemed to have been checked already.  Should the case of "you could have a French country with a Belgian VAT. " play a role anyways @tsb-odoo